conanfile.py

```diff
@@ -251,6 +253,7 @@ def _configure_cmake(self):
             self._cmake.definitions[opt.upper()] = self.options.get_safe(opt, False)
         if self.options.enable_testing:
             self._enable_error_injectors()
+        self._cmake.definitions["CELIX_ERR_BUFFER_SIZE"] = self.options.celix_err_buffer_size
```

Comment:
As discussed in https://github.com/conan-io/conan/issues/8136, Conan treat every option as string.
It will help a lot to raise exception in `validate()` if a non-number value is provided.